pkg/api/v1alpha1/cluster.go

```diff
@@ -740,3 +759,39 @@ func validatePodIAMConfig(clusterConfig *Cluster) error {
 	}
 	return nil
 }
+
+func validateCPUpgradeRolloutStrategy(clusterConfig *Cluster) error {
+	if clusterConfig.Spec.ControlPlaneConfiguration.UpgradeRolloutStrategy == nil {
+		fmt.Println("ControlPlaneConfiguration: UpgradeRolloutStrategy not specified in cluster config. CAPI will default to 'RollingUpdate' with maxSurge=1")
+		return nil
+	}
+
+	if clusterConfig.Spec.ControlPlaneConfiguration.UpgradeRolloutStrategy.Type != "RollingUpdate" {
+		return fmt.Errorf("ControlPlaneConfiguration: only 'RollingUpdate' supported for upgrade rollout strategy type")
+	}
+
+	if clusterConfig.Spec.ControlPlaneConfiguration.UpgradeRolloutStrategy.RollingUpdate.MaxSurge > 1 {
+		return fmt.Errorf("ControlPlaneConfiguration: maxSurge for control plane must be 0 or 1")
+	}
+
+	return nil
+}
+
+func validateMDUpgradeRolloutStrategy(clusterConfig *Cluster) error {
```

Comment:
don't we need to check for provider support here, if only Tinkerbell supports it

Reply:
This would be provider-wide. It is generic capi functionality. I only have unit tests for tinkerbell right now, but I am going to add similar unit test in other providers. Currently these knobs default to the values that CAPI already has.